src/qforte/ucc/spqe.py

```diff
@@ -101,9 +102,41 @@ def run(self,
             if occupation:
                 self._nbody_counts.append(0)
 
+        # create a pool of particle number, Sz, and spatial symmetry adapted second quantized operators
+        ref = sum([b << i for i, b in enumerate(self._ref)])
+        mask_alpha = 0x5555555555555555
+        mask_beta = mask_alpha << 1
+        nalpha = sum(self._ref[0::2])
+        nbeta = sum(self._ref[1::2])
+
+        idx = 0
+        # determinant id : excitation operator index
+        self._excitation_dictionary = {}
+        # list that holds the ids of determinants corresponding to operators in the pool
+        self._excitation_indices = []
         self._pool_obj = qf.SQOpPool()
-        for I in range(2 ** self._nqb):
-            self._pool_obj.add_term(0.0, self.get_op_from_basis_idx(I))
+        for I in range(1 << self._nqb):
+            alphas = [int(j) for j in bin(I & mask_alpha)[2:]]
```

Comment:
Why is the `[2:]` necessary?

Reply:
Let's assume that we have M spin-orbitals. Using `& mask_alpha` on a given M-bit string will return an M-bit string with 0 in the "beta" positions and the alpha positions unchanged. I want the "alphas" list to have length equal to the number of spatials (M/2), hence the `[2:]` part.

Reply:
I just tested this. I think you mean that you want to strip the "0b" prefix?

Reply:
You are correct. I thought I was looping every two elements, but I was wrong.

Reply:
That'd be [::2], so not too far off.